install/src/Env.hs

```diff
@@ -40,20 +42,32 @@ existsExecutable executable = liftIO $ isJust <$> findExecutable executable
 isWindowsSystem :: Bool
 isWindowsSystem = os `elem` ["mingw32", "win32"]
 
+-- | Defines all different hie versions that are supported.
+-- On windows, `8.6.3` is excluded as this version of ghc does not work there
+supportedGhcVersions :: [VersionNumber]
+supportedGhcVersions = sort (commonVersions ++ osVersions)
+  where commonVersions = ["8.4.2", "8.4.3", "8.4.4", "8.6.1", "8.6.2", "8.6.4", "8.6.5"]
```

Comment:
hard coding makes it more complex to update the range of supported ghc versions. Currently, supported ghc versions are determined by `stack-*.yaml` files. This makes it easy to add or remove supported versions, no change of install.hs necessary. Only OS dependant exceptions have to be specified.

Reply:
Well, i like in general to have the constraints explicitly in the code to being able to generate/validate/check conventions (like stack yaml file names).
This way we could validate that the stack yaml files names are correct (or even if the compiler version used by the resolver is).
The pr doesnt validate the stack yaml files but it filters out the stack files which versions are not in the explicit list (so if we dont remember to delete it, it would not be used in the script)

That said, if we found that, in this case, the possible benefits of a explicit list doesnt worth the duplication with the stack yaml files i would remove it

Reply:
Good point. Let's wait for other opinions, e.g. @alanz, @power-fungus and @bubba.

Reply:
FWIW I don't have a strong opinion on this. I do think capturing this info in one place is worthwhile, but if you are doing a stack install you need the appropriate stack file, so that does seem to work.

The only problem is if it leaves cabal out i the cold.

Another option might be to do it something like what [haskell-ci](https://github.com/haskell-CI/haskell-ci) does, taking the versions from the cabal file.

Reply:
Mmm, use an existing field (`tested-with`) of a standard (in the haskell world) config file sounds good. 
However `haskell-ci` depends on `Cabal` to parse the `.cabal` file. We will need to add it or use an ad-hoc parser to extract the info of `tested-with` and i dont fully like neither of both

Reply:
I don't have a strong opinion on this too, but explicit list sounds more preferable.

Reply:
In case my opinion counts, I'm against duplication so enumerating *.yml files sounds more preferable. We already have too many build-related reports.

And if we can forget something - then it's a good idea to check that with CI. So if you decide to leave the explicit list you should check during CI that the list is consistent with the `stack-*.yml` files.

Reply:
Ok, as i dont have neither an strong opnion on this i will use the list of `stack-<ghc-version>.yaml`

Reply:
@nponeccop Of course, your opinion counts! thank you for your input!

Reply:
done!